lib/resourceapply/core.go

```diff
@@ -57,6 +57,8 @@ func ApplyService(client coreclientv1.ServicesGetter, required *corev1.Service)
 	resourcemerge.EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 	selectorSame := equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector)
 	typeSame := equality.Semantic.DeepEqual(existing.Spec.Type, required.Spec.Type)
+	resourcemerge.EnsureServicePorts(modified, &existing.Spec.Ports, required.Spec.Ports)
```

Comment:
nit: instead of dropping this between the `selectorSame` and `typeSame` definitions and the if-statement consuming it, can we put this right after `EnsureObjectMeta` a few lines up?